src/library/scala/collection/Seq.scala

```diff
@@ -266,8 +266,10 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     *  @return `true` if this $coll has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = {
+    if(length - that.size < 0) return false
```

Comment:
For an easy guard, avoid `return` (which raises hackles) and prefer:
```
def f: Boolean = if (condition) false else {
  //body
}
```
which keeps current indentation, which still emphasize what the usual case is.

Reply:
```suggestion
    if(this.sizeCompare(that) < 0) false else {
```

Reply:
I've implemented the guard with Julien's suggestion. Thanks for the tip!